termuic/termuiConsole.go

```diff
@@ -0,0 +1,276 @@
+	tc.TermuiConsoleMetrics.Store(core.MetricNonce, 0)
+	tc.TermuiConsoleMetrics.Store(core.MetricNumConnectedPeers, 0)
+	tc.TermuiConsoleMetrics.Store(core.MetricSynchronizedRound, 0)
+
+	tc.TermuiConsoleMetrics.Store(core.MetricPublicKey, "")
+	tc.TermuiConsoleMetrics.Store(core.MetricShardId, 0)
+	tc.TermuiConsoleMetrics.Store(core.MetricTxPoolLoad, 0)
+
+	tc.TermuiConsoleMetrics.Store(core.MetricCountConsensus, 0)
+	tc.TermuiConsoleMetrics.Store(core.MetricCountLeader, 0)
+	tc.TermuiConsoleMetrics.Store(core.MetricCountAcceptedBlocks, 0)
+}
+
+// SetInt64Value method - will update the value for a key
+func (tc *TermuiConsole) SetInt64Value(key string, value int64) {
+	if _, ok := tc.TermuiConsoleMetrics.Load(key); ok {
+		tc.TermuiConsoleMetrics.Store(key, int(value))
+	}
+}
+
+// SetUInt64Value method - will update the value for a key
+func (tc *TermuiConsole) SetUInt64Value(key string, value uint64) {
+	if _, ok := tc.TermuiConsoleMetrics.Load(key); ok {
+		tc.TermuiConsoleMetrics.Store(key, int(value))
+	}
+}
+
+// SetStringValue method - will update the value of a key
+func (tc *TermuiConsole) SetStringValue(key string, value string) {
+	if _, ok := tc.TermuiConsoleMetrics.Load(key); ok {
+		tc.TermuiConsoleMetrics.Store(key, value)
+	}
+}
+
+// Increment - will increment the value of a key
+func (tc *TermuiConsole) Increment(key string) {
+	if keyValueI, ok := tc.TermuiConsoleMetrics.Load(key); ok {
+
+		keyValue := keyValueI.(int)
+		keyValue++
+		tc.TermuiConsoleMetrics.Store(key, keyValue)
+	}
+}
+
+// Decrement method - will decrement the value for a key
+func (tc *TermuiConsole) Decrement(key string) {
+	return
+}
+
+// Close method - won't do anything
+func (tc *TermuiConsole) Close() {
+	return
+}
+
+// Start method - will start termui console
+func (tc *TermuiConsole) Start() {
+	go func() {
+		if err := ui.Init(); err != nil {
+
+		}
+		defer ui.Close()
+
+		tc.eventLoop()
+	}()
+
+}
+
+func (tc *TermuiConsole) eventLoop() {
+
+	tc.termuiConsoleWidgets = NewtermuiConsoleGrid()
+
+	time.Sleep(1 * time.Second)
+
+	grid := tc.configConsoleWithNormalTermuiWidgets()
+
+	uiEvents := ui.PollEvents()
+	// handles kill signal sent to gotop
+	sigTerm := make(chan os.Signal, 2)
+	signal.Notify(sigTerm, os.Interrupt, syscall.SIGTERM)
+
+	drawTicker := time.NewTicker(time.Second).C
+
+	for {
+		select {
+		case <-drawTicker:
+			tc.refreshDataForConsole()
+
+			ui.Clear()
+			ui.Render(grid)
+		case <-sigTerm:
+			ui.Clear()
+
+			return
+		case e := <-uiEvents:
+			switch e.ID {
+			case "<Resize>":
+				payload := e.Payload.(ui.Resize)
+				grid.SetRect(0, 0, payload.Width, payload.Height)
+				ui.Clear()
+				ui.Render(grid)
+			case "q":
+				ui.Clear()
+				grid = tc.changeConsoleDisplay()
+
+			case "<C-c>":
+				ui.Clear()
+				_ = syscall.Kill(syscall.Getpid(), syscall.SIGINT)
+
+				return
+			}
+		}
+	}
+}
+
+func (tc *TermuiConsole) refreshDataForConsole() {
+	if tc.viewBigLog == false {
```

Comment:
a function that does 2 things by reading a flag. Apply clean code patterns here.

Reply:
done.